typing/typecore.ml

```diff
@@ -1537,6 +1543,7 @@ let partial_pred ~lev ?mode ?explode env expected_ty constrs labels p =
   let state = save_state env in
   try
     reset_pattern None true;
+    total_pat_split := 1000;
```

Comment:
This is an arbitrary constant to make sure that search does not go on forever. There is no relation between this and existentials.
It was chosen based on the minichess example, but I'm not sure it's a good idea. In practice, it should not come into play that often.

Reply:
Concerning existentials, the backtracking of `current_time` means that we shall never overflow in practice, so the error message there is just an extra security.

Reply:
> There is no relation between this and existentials.
It was chosen based on the minichess example,  but I'm not sure it's a good idea.

Ack.
Do you think we could find a value that would allow to revert the diff in `tests/typing-warnings/exhaustiveness.ml`?

> Concerning existentials, the backtracking of `current_time` means that we shall never overflow in practice, so the error message there is just an extra security.

I agree with that but I think it's a good idea too keep it (you never know what crazy code people might generate).